esmf-semantic-aspect-meta-model/src/main/resources/samm/characteristic/2.3.0/characteristic-shapes.ttl

```diff
@@ -0,0 +1,869 @@
+        sh:prefixes samm:prefixDeclarations ;
+        sh:select """
+           select $this ?values ?dataType ?item ?definedDataType ?code ?highlight
+           where {
+             $this rdf:type/rdfs:subClassOf* samm-c:Enumeration .
+             $this samm-c:values ?values .
+             $this samm:dataType ?definedDataType .
+             ?values rdf:rest*/rdf:first ?item .
+             optional { ?item a samm:Value ; samm:value ?innerValue . }
+             bind( coalesce( datatype(?innerValue), datatype(?item) ) AS ?literalDataType ) .
+             filter( strlen(str(?literalDataType)) > 0 ) .
+             filter( ?literalDataType != ?definedDataType ) .
+             bind( 'ERR_WRONG_DATATYPE' as ?code )
+             bind( ?item as ?highlight )
+           }
+        """
+     ] ;
+     sh:sparql [
+        a sh:SPARQLConstraint ;
+        sh:message "EnumerationShape '{$this}': DataType is a literal type but one of the values ('{?item}') is defined as samm:Property." ;
+        sh:prefixes samm:prefixDeclarations ;
+        sh:select """
+           select $this ?values ?item ?code ?highlight
+           where {
+              $this rdf:type/rdfs:subClassOf* samm-c:Enumeration .
+              filter ( not exists { $this samm:dataType samm:Property } ) .
+              $this samm-c:values ?values .
+              ?values rdf:rest/rdf:first ?item .
+              filter ( exists { ?item rdf:type samm:Property } )
+              bind( 'ERR_LITERAL_EXPECTED' as ?code )
+              bind( ?item as ?highlight )
+           }
+        """
+     ] ;
+     sh:sparql [
+        a sh:SPARQLConstraint ;
+        sh:message "EnumerationShape '{$this}' contains value: '{?value}' (count: {?count})" ;
+        sh:prefixes samm:prefixDeclarations ;
+        sh:select """
+           select distinct $this ?value ?count
+           where {
+             {
+               select $this ?value (count(?value) AS ?count)
+               where {
+                 $this samm-c:values ?list .
+                 ?list rdf:rest*/rdf:first ?value .
+               }
+               group by $this ?value
+               having (?count > 1)
+             }
+           }
+        """
+     ] ;
+     sh:sparql [
+        a sh:SPARQLConstraint ;
+        sh:message "EnumerationShape '{$this}' contains samm:Value: '{?value}' (count: {?count})" ;
+        sh:prefixes samm:prefixDeclarations ;
+        sh:select """
+          select distinct $this ?value ?count
+          where {
+            {
+              select $this ?value (count(?value) AS ?count)
+              where {
+                $this samm-c:values ?list .
+                ?list rdf:rest*/rdf:first ?node .
+                ?node a samm:Value ;
+                      samm:value ?value .
+              }
+              group by $this ?value
+              having (?count > 1)
+            }
+          }
+        """
+      ] ;
+   ] .
```

Comment:
Delete the second SPARQLConstraint (it's covered by `/samm:value?` part in the node path of the first shape)